### PR TITLE
i18n(it): Modified frontmatter.md and overrides.md

### PR DESCRIPTION
**Pull request by casungo**
#### Description

Updated and translated `frontmatter.md` and `overrides.md` so now all of the files in the reference folder are more true to the english version.

### CODE REVIEW

File: docs/src/content/docs/it/reference/frontmatter.md [hunk 1]
```diff
@@ -272,13 +317,14 @@ sidebar:
 
 #### `hidden`
 
-**type:** `boolean`
-**default:** `false`
+**tipo:** `boolean`
```
Comment (HiDeoo) [inline docs/src/content/docs/it/reference/frontmatter.md:320]:
```suggestion
**tipo:** `boolean`  
```

Small tweak to get the type and default value on different lines.

File: docs/src/content/docs/it/reference/frontmatter.md [hunk 2]
```diff
@@ -272,13 +317,14 @@ sidebar:
...
+**predefinito:** `false`
 
 Impedisce che questa pagina venga inclusa in un gruppo della barra laterale generato automaticamente.
 
 ```md
 ---
+# src/content/docs/example.md
 title: Pagina da nascondere dalla barra laterale generata automaticamente
 sidebar:
   hidden: vero
```
Comment (HiDeoo) [inline docs/src/content/docs/it/reference/frontmatter.md:330]:
```suggestion
  hidden: true
```

Not related to your changes but should not be translated.

File: docs/src/content/docs/it/reference/frontmatter.md [hunk 3]
```diff
@@ -314,16 +362,85 @@ sidebar:
...
+};
+```
+
+Per saperne di più sugli schemi di raccolta dei contenuti, consulta [“Definizione di uno schema di raccolta”](https://docs.astro.build/it/guides/content-collections/#defining-a-collection-schema) nella documentazione di Astro.
+
+`docsSchema()` accetta le seguenti opzioni:
+
+### `extend`
+
+**tipo:** Schema Zod o funzione che restituisce uno schema Zod
```
Comment (HiDeoo) [inline docs/src/content/docs/it/reference/frontmatter.md:400]:
```suggestion
**tipo:** Schema Zod o funzione che restituisce uno schema Zod  
```

Same as above to spread the type and default value on 2 lines.

**Review by HiDeoo — CHANGES_REQUESTED** [review]
Thanks for the translation :raised_hands:

As a non-native speaker, I reviewed this PR using the [Docs Translation Review Checklist](https://hackmd.io/1KYnrpUQR0SAGTzHIUoUDg?view).

I left a few suggestions, mostly aesthetic, but except for that, this looks great to me and we should be able to merge this soon :thumbsup:

**Comment by casungo** [comment]
Thank you for your review! I pushed a commit now, should be everything :)

**Review by HiDeoo — APPROVED** [review]
Thanks for the quick update, this looks perfect to me :raised_hands:

**Pull request merged**